packages/astrocite-eutils/src/eutils.ts

```diff
@@ -12,11 +12,54 @@ const FIELD_TRANSFORMS = new Map<
     [
         'articleids',
         ({ articleids, uid }) => {
-            return {
-                URL: articleids.some(({ value }) => value === uid)
-                    ? `https://www.ncbi.nlm.nih.gov/pubmed/${uid}`
-                    : `https://www.ncbi.nlm.nih.gov/pmc/articles/PMC${uid}`,
+            const data: Partial<Data> = {};
+
+            const chooseURL = () => {
+                const pmcid = articleids.find(({ idtype, value }) => {
+                    return idtype === 'pmcid' && value === `PMC${uid}`;
+                });
+
+                if (pmcid) {
+                    return `https://www.ncbi.nlm.nih.gov/pmc/articles/${pmcid.value}`;
+                }
+
+                const pmid = articleids.find(({ idtype, value }) => {
+                    return idtype === 'pubmed' && value === uid;
+                });
+
+                if (pmid) {
+                    return `https://www.ncbi.nlm.nih.gov/pubmed/${pmid.value}`;
+                }
+
+                return undefined;
             };
+
+            const url = chooseURL();
+
+            if (url) {
+                data.URL = url;
+            }
```

Comment:
Is any of this even necessary? Can't the same thing be done in the for loop below without having to loop through extra times?

For example... If, for instance, case `pmc` or `pmcid` hits, couldn't we just assume the URL for the publication is just `https://www.ncbi.nlm.nih.gov/pmc/articles/${pmcid.value}` and set it accordingly?

Reply:
It's tricky, because the `pmcid`, `pmc`, `pmid` and `pubmed` articleid values are different in eSummary responses when the db is either "pubmed"  or "pmc".

I've had another look, and it _might_ be safe to say that if there's a "pubmed" articleid (only present in "pubmed" responses) then use that, otherwise if there's a "pmcid" articleid (present in both responses, but only with a useful value in "pmc" responses), then use that.

Reply:
The code in the PR is more explicit, by comparing those values to what's expected based on the uid.

Instead of looping through several times, it could perhaps build an object containing the articleid keys and values in a single step.

Reply:
> It's tricky, because the pmcid, pmc, pmid and pubmed articleid values are different in eSummary responses when the db is either "pubmed" or "pmc".

Do you have an example of this? Sorry, it's been a while since I've dealt directly with the API response and there's a few cobwebs.

Reply:
Updated to process the identifiers in a single loop.

Reply:
* `db=pubmed`: https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pubmed&id=28507505&retmode=json
* `db=pmc`: https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pmc&id=5410569&retmode=json